src/main/java/Duke/InputParser.java

```diff
@@ -0,0 +1,96 @@
+package Duke;
+
+public class InputParser {
+
+    private String userCommand;
+    private String inputBuffer;
+
+    public InputParser() {
+        inputBuffer = "";
+        userCommand = "";
+    }
+
+    private void clear() {
+        inputBuffer = "";
+        userCommand = "";
+    }
+
+    private boolean isValidCommand() {
+        switch (userCommand) {
+            case ("list"): //Fallthrough
+            case ("mark"): //Fallthrough
+            case ("unmark"): //Fallthrough
+            case ("todo"): //Fallthrough
+            case ("deadline"): //Fallthrough
+            case ("event"):
+                return true;
+            default:
+                return false;
+        }
+    }
+    private boolean isCorrectInput(String[] parsed ) {
+        if (userCommand.equals("list")){
+            return true;
+        }
+        return (parsed.length > 1);
+    }
+
+    private String[] parseParameter(String inputString, String optionFlag){
+        int OPTION_LEN = 4;
+        int optionIndex = inputString.indexOf(optionFlag);
+
+        String descriptionMain = inputString.substring(0, optionIndex);
+        String descriptionOption = inputString.substring(optionIndex + OPTION_LEN);
+
+        return new String[]{ descriptionMain , descriptionOption };
+    }
+
+    public void parseUserInput(String userInput) throws UnknownCommandException, DukeException {
+
+        String[] inputSplitBySpace = userInput.split(" ", 2);
```

Comment:
Do consider avoiding the usage of magic literals. Store and use the values as `final` variables instead